articles/file-handling-in-Cpp/index.md

```diff
@@ -0,0 +1,176 @@
+<!--more-->
+### Why do you need File Handling?
+- **Reusability**: Generally, developers work with big data, and it's always not efficient to write everything out by storing data in files, one can use those files anytime.
+
+- **Larger Storage Capacity**: Storing a large amount of data or test cases (to make sure that your code runs in every condition) is a tedious task, and by using files, a developer can store that data easily.
+
+- **Portability**: With files, developers can transfer the data without worrying about data loss.
+
+In this article, we will walk through how to implement a file handling function in C++, and we will cover the key properties and challenges when it comes to file handling.
+
+- Opening a File
+- Writing to a File
+- Reading from a File
+- Closing and Saving a file
+
+To perform file handling functions, we need to add another library: fstream. This library holds all the functions related to handling files. Here's how to include it.
+
+`#include<fstream>`
+
+Let's dive straight into learning.
+
+### Opening a File
+One of the most common actions in File Handling is opening a file. Let's see how to do it:
+
+```C
+ifstream fin;
+fin.open("index.txt");
+```
+
+That was simple as we just used the function **open** to open a file name "index.txt." Here we use the ` ifstream` object to open our file.
+
+**ifstream** is the type we use to declare an object. Very similar to the datatypes that we use in other functions, such as: int, float, bool or string.
+
+**fin** is the name of the variable. We could name it anything we want.
+
+Note: The one thing to keep in mind is that if the file we are trying to open doesn't exist on our computer, then this programs automatically create a new one with the same name or delete everything in it if something does exist in it. We can also opt this option whenever we need to create a file. You can also give a second argument to specify how you want to handle your file:
+* ios::app   -- Append to the file
+* ios::trunc -- Delete everything in the file
+
+In case, you want to append the file. Here's how you would do it:
+```C
+ifstream fin;
+fin.open("index.txt", ios::app);
+```
+
+You won't always know what file to open, so we can also opt-in for this option for asking the name of the file. Here's how we can do it:
+
+```C
+string filename;
+cout<<"Enter the name of the File?";
+cin>>filename
+ifstream fin;
+fin.open(filename.data());
+```
+
+Here in this code, we just asked for the name of the file, and proceed with the same steps we did above.
+
+One of the most common errors we face while opening a File is; what if a file is already open before we are using it. We need to ensure that doesn't happen because otherwise, we would not be able to open it, and our file can get corrupted, and we may lose all of our existing data. Here's how we can ensure that doesn't happen:
```

Comment:
opening a File is; what if a file -> opening a File is: what if a file

Reply:
happen because otherwise, we would not be able -> happen because otherwise we would not be able

Reply:
able to open it, and our file can get corrupted, and we may lose all  ->  able to open it, our file can get corrupted, and we may lose all